flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/DeactivatedCheckpointCompletedCheckpointStore.java

```diff
@@ -69,6 +69,11 @@ public SharedStateRegistry getSharedStateRegistry() {
         throw unsupportedOperationException();
```

Comment:
Yes, the javadoc is wrong (though it wasn't changed in this PR). 
It doesn't sound like it **must** throw an exception instead of NO_OP.

Anyways, I can fix javadoc and also list `getRegistry` as an allowed method in it (I'm afraid that throwing an exception or returning throwing `SharedStateRegistry` can break something).
WDYT?

Reply:
Yes, the javadoc is wrong (though it wasn't changed in this PR). 
It doesn't sound like it **must** throw an exception instead of NO_OP.

Anyways, I can fix javadoc and also list `getRegistry` as an allowed method in it (I'm afraid that throwing an exception or returning throwing `SharedStateRegistry` can break something).
WDYT?

Reply:
For the sake of consistency and safety I think it would be better to have it throwing an exception. If this class shouldn't be used (and that's the intention), then better make sure it won't be used at least on the runtime level. Also it seems it should be working fine with throwing version (based on the usages). Or at least not worse than it used to.

For the record, I don't like the throwing version. Ideally with checkpointing disabled, this class should just not be instantiated (for example even having field `SchedulerBase#completedCheckpointStore` as `Optional` or `@Nulalble` would be better), but that's out of scope of this ticket.